plugin/trino-exasol/src/main/java/io/trino/plugin/exasol/ExasolClient.java

```diff
@@ -0,0 +1,290 @@
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, Connection connection, String catalogName, String remoteSchemaName, String remoteTableName, String newRemoteSchemaName, String newRemoteTableName)
+    {
+        if (!remoteSchemaName.equals(newRemoteSchemaName)) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables across schemas");
+        }
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables");
+    }
+
+    @Override
+    protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCondition joinCondition)
+    {
+        // Deactivated because test 'testJoinPushdown()' requires write access which is not implemented for Exasol
+        return false;
+    }
+
+    @Override
+    public Optional<JdbcExpression> implementAggregation(ConnectorSession session, AggregateFunction aggregate, Map<String, ColumnHandle> assignments)
+    {
+        // Deactivated because test 'testCaseSensitiveAggregationPushdown()' requires write access which is not implemented for Exasol
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        switch (typeHandle.jdbcType()) {
+            case Types.BOOLEAN:
+                return Optional.of(booleanColumnMapping());
+            case Types.SMALLINT:
+                return Optional.of(smallintColumnMapping());
+            case Types.INTEGER:
+                return Optional.of(integerColumnMapping());
+            case Types.BIGINT:
+                return Optional.of(bigintColumnMapping());
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
+            case Types.DECIMAL:
+                int decimalDigits = typeHandle.requiredDecimalDigits();
+                int columnSize = typeHandle.requiredColumnSize();
+                return Optional.of(decimalColumnMapping(createDecimalType(columnSize, decimalDigits)));
+            case Types.CHAR:
+                return Optional.of(defaultCharColumnMapping(typeHandle.requiredColumnSize(), true));
+            case Types.VARCHAR:
+                // String data is sorted by its binary representation.
+                // https://docs.exasol.com/db/latest/sql/select.htm#UsageNotes
+                return Optional.of(defaultVarcharColumnMapping(typeHandle.requiredColumnSize(), true));
+            case Types.DATE:
+                return Optional.of(dateColumnMapping());
+            case Types.TIMESTAMP:
```

Comment:
Is a `TIMESTAMP(3) WITH LOCAL TIME ZONE` also reported as same SqlType? Or is a `TIMESTAMP(3) WITH LOCAL TIME ZONE ` just an "signal" and actual stored value is still `TIMESTAMP(3)` only?

Reply:
Yes, `TIMESTAMP(3) WITH LOCAL TIME ZONE` is reported as JDBC type `TIMESTAMP = 93` and type name `TIMESTAMP`.